acorn-loose/src/state.js

```diff
@@ -157,9 +159,23 @@ export class LooseParser {
     return this.parseTopLevel()
   }
 
+  hasPlugin(pluginName) {
+    return pluginList.indexOf(pluginName) > -1
+  }
+
   static extend(...plugins) {
     let cls = this
-    for (let i = 0; i < plugins.length; i++) cls = plugins[i](cls)
+    for (let i = 0; i < plugins.length; i++) {
+      const pluginImpl = plugins[i](cls)
+      if (typeof pluginImpl === "object") {
```

Comment:
Again, why is this new plugin format needed?

Reply:
As the issue said,this format is more convenient for the plugin author to check the acorn instance has been loaded some other proposal plugin. If using the function name that could not be exactly when other plugin check if other plugin loaded, may be the plugin author that provide another name differ from the proposal.

Reply:
How does this format help for that, though? The architecture already gets plugins by name, meaning the code that calls them has the name—why do they also have to return it?

Reply:
Sorry... I typed an error. In my thought, we can use the object that plugin return to extend another field called `name` to store the proposal name, that can easy to check whether a plugin has been loaded.